sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -887,20 +900,52 @@ def upload_manifest(
                 name=repository,
                 reference=tag_or_digest,
                 payload=data,
-                content_type=OCI_MANIFEST_MEDIA_TYPE,
-                headers={"Accept": OCI_MANIFEST_MEDIA_TYPE},
+                content_type=media_type,
                 cls=_return_response_headers,
                 **kwargs
             )
             digest = response_headers['Docker-Content-Digest']
             if not _validate_digest(data, digest):
-                raise ValueError("The digest in the response does not match the digest of the uploaded manifest.")
+                raise ValueError("The digest in the response does not match the digest of the set manifest.")
         except Exception as e:
             if repository is None or manifest is None:
                 raise ValueError("The parameter repository and manifest cannot be None.") from e
             raise
         return digest
 
+    @distributed_trace
+    def get_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> GetManifestResult:
+        """Get the manifest for an artifact.
+
+        :param str repository: Name of the repository.
+        :param str tag_or_digest: The tag or digest of the manifest to get.
+            When digest is provided, will use this digest to compare with the one calculated by the response payload.
+            When tag is provided, will use the digest in response headers to compare.
+        :returns: GetManifestResult
+        :rtype: ~azure.containerregistry.GetManifestResult
+        :raises ValueError: If the requested digest does not match the digest of the received manifest.
+        """
+        response, manifest_iterator = cast(
+            Tuple[PipelineResponse, Iterator[bytes]],
+            self._client.container_registry.get_manifest(
+                name=repository,
+                reference=tag_or_digest,
+                accept=SUPPORTED_MANIFEST_MEDIA_TYPES,
+                cls=_return_response_and_deserialized,
+                **kwargs
+            )
+        )
+        media_type = response.http_response.headers['Content-Type']
+        manifest_bytes = b"".join(manifest_iterator)
+        if tag_or_digest.startswith("sha256:"):
+            digest = tag_or_digest
+        else:
+            digest = response.http_response.headers['Docker-Content-Digest']
+        if not _validate_digest(manifest_bytes, digest):
+            raise ValueError("The requested digest does not match the digest of the received manifest.")
+
+        return GetManifestResult(digest=digest, manifest=_deserialize_manifest(manifest_bytes), media_type=media_type)
```

Comment:
Rather than use this `deserialize` function - we should be able to pull the JSON directly from the response object

Reply:
The data we can get from response object is encoded bytes. I don't find JSON content in response object.

Reply:
This method is on the response - we should use it load the JSON rather than do it ourselves:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/rest/_http_response_impl.py#L283-L294

Reply:
Updated